sphinxcontrib/needs/filter_common.py

```diff
@@ -80,16 +75,12 @@ def process_filters(all_needs, current_needlist):
 
     sort_key = current_needlist["sort_by"]
     if sort_key:
-        if sort_key == "status":
-            all_needs = sorted(all_needs, key=status_sorter)
-        else:
-            try:
-                sorted_needs = sorted(all_needs, key=lambda node: node[sort_key])
-                all_needs = sorted_needs
-            except Exception as e:
-                logger.warning(
-                    "Sorting parameter {0} not valid: Error: {1}".format(sort_key, e)
-                )
+        try:
+            all_needs = sorted(all_needs, key=lambda node: node[sort_key])
```

Comment:
what if a value is None?
then sorted will throw an exception as comparison between None and string does not work

Reply:
Maybe we should have a test case, where certain fields are set and not set.
Then test the here changed code.

It all may look simpler now,  but the checks are needed if you can't "trust" the incoming data.
I personal don't like to trust data validation on upper levels only.
The upper level may not know, what kind of validation is needed here and it even may change.
Python version change, Sphinx version , ....
So my goal is to be able to sort here everything, not matter how it looks like. For status at least: "abc", "", " " and None.

Reply:
i'll revert this change.

i have to respectfully disagree with putting the type checking at the 'lower level'. It would be far better to enforce the typing at the API boundary, so that your code-base only has to deal with sanitised data. that way any type errors are caught by a linter (like mypy), rather than with messy type-checks in every function.

the way you'd do this is by changing the 'node info' from a stringly-typed `dict` to a strongly-typed concrete type (then you would *know* the type of `node[sort_key]`, without needing to check). that's a big change, and a problem for another day!